File: clients/python/model/explosion_params.py
class ExplosionParams:
    def __init__(self, radius, damage):
        self.radius = radius
        self.damage = damage
    @staticmethod
    def read_from(stream):
        radius = stream.read_double()
        damage = stream.read_int()
        return ExplosionParams(radius, damage)
    def write_to(self, stream):
        stream.write_double(self.radius)
        stream.write_int(self.damage)
    def __repr__(self):
        return "ExplosionParams(" + \
            repr(self.radius) + "," + \
            repr(self.damage) + \
            ")"

File: clients/ruby/model/explosion_params.rb
class ExplosionParams
    attr_accessor :radius
    attr_accessor :damage
    def initialize(radius, damage)
        @radius = radius
        @damage = damage
    end
    def self.read_from(stream)
        radius = stream.read_double()
        damage = stream.read_int()
        ExplosionParams.new(radius, damage)
    end
    def write_to(stream)
        stream.write_double(@radius)
        stream.write_int(@damage)
    end
end
